app/controllers/application_controller.rb

```diff
@@ -1,10 +1,149 @@
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
-
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, 'secret'
+  end
+
+  get '/' do
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :index
+    end
+  end
+  ####################### signup flow ##############
+  get '/signup' do
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params.any? { |_k, v| v.empty? }
+      redirect '/signup'
+    else
+      @user = User.create(params)
+      session[:id] = @user.id
+      redirect to '/tweets'
+    end
+  end
+  ################# log in flow ###################
+  get '/login' do
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(params)
+    if @user
+      session[:id] = @user.id
+      redirect to '/tweets'
+    else
+      redirect to '/signup'
+    end
+  end
+  ##################################################
+
+  get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).